helix-core/src/words.rs

```diff
@@ -0,0 +1,33 @@
+use crate::movement::{categorize, is_horiz_blank, is_word, skip_over_prev};
+use ropey::RopeSlice;
+
+#[must_use]
+pub fn nth_prev_word_boundary(slice: RopeSlice, mut char_idx: usize, count: usize) -> usize {
```

Comment:
Could you reuse `move_prev_word_start` and take `range.head`?

Reply:
But that would do some extra unnecessary stuff though, we don't need end.

Reply:
It does a few extra iterations to find the start of the word, but it's unlikely to be a significant performance difference, and it reduces code duplication. I'm undecided here, thought it would be a good idea since we already had some bugs to fix in the `move_*_word` functions

Reply:
I personally find this better, because this is quite separated from that, as in this only looks for the last word. Maybe to be safe I just add a test for this?

Reply:
Test is OK 👍